src/rulesets/Base/indexing.jl

```diff
@@ -27,3 +27,50 @@ function rrule(::typeof(getindex), x::Array{<:Number}, inds...)
     return y, getindex_pullback
 end
 
+#####
+##### `eachslice` and friends
+#####
+
+function rrule(::typeof(eachrow), x::AbstractVecOrMat)
+    allrows(dy) = (NoTangent(), ∇eachslice(unthunk(dy), x, Val(1)))
+    return collect(eachrow(x)), allrows
+end
+
+function rrule(::typeof(eachcol), x::AbstractVecOrMat)
+    allcols(dy) = (NoTangent(), ∇eachslice(unthunk(dy), x, Val(2)))
+    return collect(eachcol(x)), allcols
+end
+
+function rrule(::typeof(eachslice), x::AbstractArray; dims)
+    y = collect(eachslice(x; dims=dims))
+    dim = if length(dims) == 1
+        only(dims)
+    else
+        throw(ArgumentError("""That's amazing, after many years JuliaLang/julia#32310 actually landed.
```

Comment:
LOL.
idk how i feel about this code.
This is basically an assertion.

Reply:
Heh. Notice that it's below `eachslice`, and so is at present unreachable. You will get *some* error if you use this rule with the new behaviour of the new eachslice, when that lands. The choice is only whether to leave it be... I guess a methoderror somewhere.

Reply:
I think if we change it to an `@assert` I am happy

Reply:
Done.